api/src/modules/translations/services/translations.service.ts

```diff
@@ -80,19 +106,43 @@ export class TranslationsService implements OnApplicationBootstrap {
 		value: string
 	): Promise<void> {
 		try {
-			await this.dataService.execute<
-				UpdateTranslationMutation,
-				UpdateTranslationMutationVariables
-			>(UpdateTranslationDocument, {
+			const existingEntries = await this.getTranslationsByComponentLocationKey(
 				component,
 				location,
-				key,
-				languageCode,
-				value,
-			});
+				key
+			);
+			if (existingEntries.find((entry) => entry.language === languageCode)) {
+				// Update entry
+				await this.dataService.execute<
+					UpdateTranslationMutation,
+					UpdateTranslationMutationVariables
+				>(UpdateTranslationDocument, {
+					component,
+					location,
+					key,
+					languageCode,
+					value,
+				});
+			} else {
+				// Insert entry (eg: for missing EN entry where NL entry already exists
+				await this.dataService.execute<
+					InsertTranslationMutation,
+					InsertTranslationMutationVariables
+				>(InsertTranslationDocument, {
+					component,
+					location,
+					key,
+					languageCode,
+					value,
+					value_type:
+						existingEntries.find((entry) => entry.language === 'NL')?.value_type ||
+						'TEXT',
+				});
+			}
```

Comment:
Split to `updateTranslation` & `insertTranslation` + rename function to `upsertTranslation`.